R/create.R

```diff
@@ -77,45 +77,94 @@ create_project <- function(path,
   invisible(TRUE)
 }
 
-#' Create a project from a github repository
+#' Create a repo and project from GitHub
+#'
+#' Creates a new local Git repository from a repository on GitHub. If you have
+#' pre-configured a GitHub personal access token (PAT) as described in
+#' [gh::gh_whoami()], you will get more sensible default behavior for the `fork`
+#' argument. You cannot create a fork without a PAT. Currently only works for
+#' public repositories. A future version of this function will likely have an
+#' interface closer to [use_github()], i.e. more ability to accept credentials
+#' and more control over the Git configuration of the affected remote or local
+#' repositories.
+#'
+#' @seealso [use_course()] for one-time download of all files in a Git repo,
+#'   without any local or remote Git operations.
 #'
 #' @inheritParams create_package
-#' @param repo Full name of repository: `owner/repo`
-#' @param fork Create a fork before cloning? Defaults to `TRUE` if you can't
-#'   push to `repo`, `FALSE` if you can.
+#' @param repo GitHub repo specification in this form: `owner/repo`. The second
+#'   part, i.e. the GitHub repo name, will be the name of the new local repo.
+#' @inheritParams use_course
+#' @param fork Create and clone a fork? Or clone `repo` itself? Defaults to
+#'   `TRUE` if you can't push to `repo`, `FALSE` if you can.
+#' @param rstudio Initiate an [RStudio
+#'   Project](https://support.rstudio.com/hc/en-us/articles/200526207-Using-Projects)?
+#'    Defaults to `TRUE` if in an RStudio session and project has no
+#'   pre-existing `.Rproj` file. Defaults to `FALSE` otherwise.
 #' @export
-create_from_github <- function(repo, path, fork = NA, open = TRUE) {
+#' @examples
+#' \dontrun{
+#' create_from_github("r-lib/usethis")
+#' }
+create_from_github <- function(repo,
+                               destdir = NULL,
```

Comment:
I'm now using `destdir` like `use_course()`. The name of the new directory = repo is determined by `repo` of `owner/repo`. So the user is really specifying a destination or parent directory. It has same default behaviour, which is to put the new repo in `~/Desktop`.

Reply:
👍